sql/hive/v2/IdentifiersParser.g4

```diff
@@ -290,12 +290,10 @@ precedenceUnaryOperator
     ;
 
 isCondition
-    : KW_NULL
-    | KW_TRUE
-    | KW_FALSE
-    | KW_NOT KW_NULL
-    | KW_NOT KW_TRUE
-    | KW_NOT KW_FALSE
+    : (KW_NOT)? KW_NULL
+    | (KW_NOT)? KW_TRUE
+    | (KW_NOT)? KW_FALSE
+    | (KW_NOT)? KW_DISTINCT KW_FROM
```

Comment:
No need to use parentheses here. Also, the common can be factored:

```suggestion
    : KW_NOT? (KW_NULL | KW_TRUE | KW_FALSE | KW_DISTINCT | KW_FROM)
```

Reply:
ok. got it